### PR TITLE
Wip 353 test dashboard b

### PR DESCRIPTION
**Pull request by LancelotMarti**

### CODE REVIEW

**Comment by PaulineMauryL** [comment]
## Coverage report


 <details><summary>Click to see where and how coverage changed</summary><table><thead>
  <tr><th>File</th><th>Statements</th><th>Missing</th><th>Coverage</th><th>Coverage<br>(new stmts)</th><th>Lines missing</th></tr>
</thead>
<tbody><tr>
<td colspan="6">&nbsp;&nbsp;<b>server/lomas_server/administration/tests</b></td><tr>
<td>&nbsp;&nbsp;<a href="https://github.com/dscc-admin-ch/lomas/pull/228/files#diff-d706b293355e5e237bc2fd0c6788e879fd9a1a15c44744f7d37a8367445e0978">test_streamlit_app_b.py</a></td>
<td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/228/files#diff-d706b293355e5e237bc2fd0c6788e879fd9a1a15c44744f7d37a8367445e0978"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/228/files#diff-d706b293355e5e237bc2fd0c6788e879fd9a1a15c44744f7d37a8367445e0978"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/228/files#diff-d706b293355e5e237bc2fd0c6788e879fd9a1a15c44744f7d37a8367445e0978"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/228/files#diff-d706b293355e5e237bc2fd0c6788e879fd9a1a15c44744f7d37a8367445e0978"></a></td><td><a href="https://github.com/dscc-admin-ch/lomas/pull/228/files#diff-d706b293355e5e237bc2fd0c6788e879fd9a1a15c44744f7d37a8367445e0978R15-R17">15-17</a>, <a href="https://github.com/dscc-admin-ch/lomas/pull/228/files#diff-d706b293355e5e237bc2fd0c6788e879fd9a1a15c44744f7d37a8367445e0978R25-R28">25-28</a>, <a href="https://github.com/dscc-admin-ch/lomas/pull/228/files#diff-d706b293355e5e237bc2fd0c6788e879fd9a1a15c44744f7d37a8367445e0978R34-R43">34-43</a>, <a href="https://github.com/dscc-admin-ch/lomas/pull/228/files#diff-d706b293355e5e237bc2fd0c6788e879fd9a1a15c44744f7d37a8367445e0978R53-R202">53-202</a>, <a href="https://github.com/dscc-admin-ch/lomas/pull/228/files#diff-d706b293355e5e237bc2fd0c6788e879fd9a1a15c44744f7d37a8367445e0978R209-R246">209-246</a></td></tbody>
<tfoot>
<tr>
<td><b>Project Total</b></td>
<td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/228/files#diff-4b0bf2efa3367c0072ac2bf1e234e703dc46b47aaa4fe9d3b01737b1a15752b1"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/228/files#diff-4b0bf2efa3367c0072ac2bf1e234e703dc46b47aaa4fe9d3b01737b1a15752b1"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/228/files#diff-4b0bf2efa3367c0072ac2bf1e234e703dc46b47aaa4fe9d3b01737b1a15752b1"></a></td><td align="center"><a href="https://github.com/dscc-admin-ch/lomas/pull/228/files#diff-4b0bf2efa3367c0072ac2bf1e234e703dc46b47aaa4fe9d3b01737b1a15752b1"></a></td><td>&nbsp;</td>
</tr>
</tfoot>
</table>

<sub>

This report was generated by [python-coverage-comment-action](https://github.com/py-cov-action/python-coverage-comment-action)

</sub>
</details>

**Comment by LancelotMarti** [comment]
Fix issues 353 and 354

File: server/run_basic_tests.sh [hunk 1]
```diff
@@ -9,6 +9,8 @@ export PYTHONPATH=$(pwd):$PYTHONPATH
 cd lomas_server/
 python -m unittest discover -s .
 ret=$?
+pytest ./administration/tests/test_streamlit_app.py
+pytest ./administration/tests/test_streamlit_app_b.py
```
Comment (damienbfs) [inline server/run_basic_tests.sh:13]: We need to get the return values back and check them

File: server/requirements_streamlit.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
+mongomock==4.3.0
```
Comment (damienbfs) [inline server/requirements_streamlit.txt:1]: Should we create a project level requirements file for running the tests and coverage?

Comment (LancelotMarti) [reply server/requirements_streamlit.txt:1]: Good point, created an issue for that (375)

File: server/lomas_server/administration/tests/test_streamlit_app_b.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+import mongomock
+import pytest
+from lomas_core.models.constants import (
+    PrivateDatabaseType,
+)
+from streamlit.testing.v1 import AppTest
+
+
+def get_mocked_db():
```
[the remark this answers is not in the document]
Comment (LancelotMarti) [reply server/lomas_server/administration/tests/test_streamlit_app_b.py]: Done, thanks

**Pull request merged**